hypershift-operator/controllers/hostedcluster/internal/platform/openstack/openstack.go

```diff
@@ -445,3 +505,39 @@ func defaultWorkerSecurityGroupRules(machineCIDRs []string) []capo.SecurityGroup
 
 	return ingressRules
 }
+
+// lookupRHCOSImage looks up a release image and extracts the RHCOS image URL and release version
+func lookupRHCOSImage(ctx context.Context, client client.Client, hcluster *hyperv1.HostedCluster) (string, string, error) {
+	// We only support amd64 for now
+	supportedArch := hyperv1.ArchitectureAMD64
+
+	var rhcosImageURL, rhcosImageRelease string
+	releaseProvider := &releaseinfo.RegistryClientProvider{}
+
+	pullSecretBytes, err := hyperutil.GetPullSecretBytes(ctx, client, hcluster)
+	if err != nil {
+		return rhcosImageURL, rhcosImageRelease, err
+	}
+
+	releaseImage, err := releaseProvider.Lookup(ctx, hcluster.Spec.Release.Image, pullSecretBytes)
+	if err != nil {
+		return rhcosImageURL, rhcosImageRelease, fmt.Errorf("failed to lookup release image: %w", err)
+	}
+
+	// We need to translate amd64 to x86_64 since that is what is in the release image stream
+	if _, ok := releaseImage.StreamMetadata.Architectures[hyperv1.ArchAliases[supportedArch]]; !ok {
+		return rhcosImageURL, rhcosImageRelease, fmt.Errorf("arch does not exist in release image, arch: %s", supportedArch)
+	}
+
+	rhcosImageURL = releaseImage.StreamMetadata.Architectures[hyperv1.ArchAliases[supportedArch]].RHCOS.OpenStackDisk.URL
```

Comment:
I'm 99% sure we can grab the SHA256s from here, too. It's probably in OpenStackDisk.